src/main/java/org/jabref/gui/citationkeypattern/GenerateCitationKeyAction.java

```diff
@@ -45,8 +48,17 @@ public void execute() {
 
         checkOverwriteKeysChosen();
 
-        BackgroundTask.wrap(this::generateKeys)
-                      .executeWith(Globals.TASK_EXECUTOR);
+        if (!this.isCanceled) {
+            Property<BackgroundTask.BackgroundProgress> backgroundProgressProperty = new SimpleObjectProperty<>(new BackgroundTask.BackgroundProgress(0, entries.size()));
```

Comment:
Have a look at the  method     public BackgroundTask<List<ImportFilesResultItemViewModel>> importFilesInBackground(List<Path> files) {

in ImportHandler
it has an updateProgress method

Reply:
Is it enough to run the update on the JavaFX Thread using the DefaultTaskExecutor (20530a2) or would it be better (cleaner) to mimic ImportHandler entirely by Inheriting from BackgroundTask?

Reply:
I would implement it similar to the import Handler method This way it's cleaner.
So you just introdue a new method which creates and retunrs the backround task.